lib/index.js

```diff
@@ -0,0 +1,300 @@
+import _get from 'lodash/get';
+import _isNil from 'lodash/isNil';
+import _merge from 'lodash/merge';
+import debugLib from 'debug';
+
+import { diffObject } from './utils/diffObject';
+
+/**
+ * Default serialize function for the server
+ * @param {object|array|null|string|number} value
+ * @return {Buffer}
+ */
+const serializeAsJSONForCache = (value) => {
+  return Buffer.from(JSON.stringify(value), 'utf8');
```

Comment:
Nit: not incredibly important but technically the `Buffer` global is not available in the browser. This function is never executed in the browser at the moment but in general a common pattern I see is providing separate entrypoints in the package for different supported execution environments. So in this case this function would live in server.js which the user would import as `import { debouncePayload } from 'braze-segment-debounce/server';` That creates a hard separation so that an environment isn't importing any code it can't execute.